main.cpp

```diff
@@ -0,0 +1,66 @@
+#include "dice.h"
+
+static void parseString(std::string &argument, Dice &obj);
+
+int main(const int argc, const char *argv[]){
+//TODO Add some basic error checking 
+    Dice roller;
+
+    if(argc <= 1){
+        std::cout << "You haven't entered any arguments" << std::endl;
+        std::cout << "The correct usage is: <number of dice>d<sides on dice>" << std::endl;
+        return -1;
+    }
+
+    for(int i = 1; i < argc; i++){
+        //I don't like this can it be cast in the function parameters instead?
+        std::string s (argv[i]);
+        parseString(s, roller);
```

Comment:
If you change parseString to take a "const std::string&", you can just pass argv[i], and it will convert it to a temporary std::string for you.